h2o-algos/src/main/java/hex/schemas/GenericV3.java

```diff
@@ -3,19 +3,18 @@
 import hex.generic.Generic;
 import hex.generic.GenericModelParameters;
 import water.api.API;
-import water.api.schemas3.KeyV3;
 import water.api.schemas3.ModelParametersSchemaV3;
 
 public class GenericV3 extends ModelBuilderSchema<Generic, GenericV3, GenericV3.GenericParametersV3> {
 
     public static final class GenericParametersV3 extends ModelParametersSchemaV3<GenericModelParameters, GenericParametersV3> {
         public static final String[] fields = new String[]{
                 "model_id",
-                "model_key"
+                "path"
         };
 
-        @API(direction = API.Direction.INOUT, level = API.Level.critical, help = "Key to the self-contained model archive already uploaded to H2O.")
-        public KeyV3.FrameKeyV3 model_key;
+        @API(required = true, level = API.Level.critical, help = "Path to file with self-contained model archive.")
```

Comment:
Removing model_key will definitely break APIs. What R and Python do is they import the file first and then use the file key to create the model.

Reply:
I have updated the R and python APIs, do we need to be backwards compatible there?

Reply:
Ok, only the tests need fixing. Will do a patch for you.